File: nattybear/level1/콜라츠-추측/Solution.java
class Solution {
  public int solution(int num) {
    long len = collatz(num);
    return (len < 500) ? (int) len : -1;
  }

  static long collatz(long x) {
    if (x == 1)
      return 0;

    long y = (x % 2 == 0) ? x / 2 : x*3 + 1;

    return 1 + collatz(y);
  }
}

File: nattybear/level1/콜라츠-추측/solution.js
function solution(num) {
  const answer  = collatz(num)
  return (answer < 500) ? answer : -1
}

function collatz(x) {
  if (x == 1)
    return 0

  if (x % 2 == 0)
    return 1 + collatz(Math.floor(x / 2))
  else
    return 1 + collatz(x*3 + 1)
}
